addons/mail/static/src/messaging/entity/user/user.js

```diff
@@ -35,38 +48,31 @@ function UserFactory({ Entity }) {
          * @override
          */
         _createInstanceLocalId(data) {
-            return `${this.constructor.name}_${data.id}`;
+            return `${this.constructor.entityName}_${data.id}`;
         }
 
         /**
          * @override
          */
-        _update(data) {
-            const {
-                displayName,
-                id = this.id,
-            } = data;
-
-            Object.assign(this, {
-                id,
-                model: 'res.user',
-            });
-
-            if (displayName) {
-                if (this.partner) {
-                    this.partner.update({ display_name: displayName });
-                } else {
-                    this._displayName = displayName;
-                }
+        _updateAfter(previous) {
+            if (this.partnerDisplayName && this.partner) {
+                this.partner.update({ display_name: this.partnerDisplayName });
```

Comment:
We really can't get rid of this? What's the problem in this case?

Reply:
sometimes we are aware of just user, sometimes just partner, and sometimes we know both. How should be store display name? In python world, this is straightforward because this is a monolith and complete architecture and it's simply on partner (and user uses related on partner), but that's not the case here (it has to assume "incompleteness").

One solution is to introduce an entity that is shared fields for partner and user. What do you think of this?